pandas/core/indexes/base.py

```diff
@@ -4425,9 +4435,28 @@ def sort_values(
         """
         idx = ensure_key_mapped(self, key)
 
-        _as = idx.argsort()
-        if not ascending:
-            _as = _as[::-1]
+        # GH 35584. Sort missing values according to na_position kwarg
+        if not isinstance(self, ABCMultiIndex):
+            bad = isna(idx)
+            good = ~bad
+            _as = np.arange(len(idx), dtype=np.int64)
+            if na_position == "last":
+                _as = np.concatenate([_as[good][idx[good].argsort()], _as[bad]])
+                if not ascending:
+                    _as[: np.sum(good)] = _as[: np.sum(good)][::-1]
+            elif na_position == "first":
+                _as = np.concatenate([_as[bad], _as[good][idx[good].argsort()]])
+                if not ascending:
+                    _as[np.sum(bad) :] = _as[np.sum(bad) :][::-1]
+            else:
+                raise ValueError(
+                    "Invalid na_position keyword argument value. "
+                    f"Can be only 'first' or 'last', '{na_position}' given."
+                )
+        else:
+            _as = idx.argsort()
```

Comment:
can you comment that this is the MI one?

it *might* be better to actually have a helper method to compute the _sort_indexer where you define for Index and MI then this method becomes simpler.

Reply:
also for MI we are basically ignoreing na_position right?

Reply:
Done.